Step1 - 용어 사전 만들기

sjhello opened the pull request:
처음 뵙겠습니다 리뷰어님!
리뷰 잘 부탁드립니다 😃

하실 말씀이 있으시다면 편하게 말씀해주세요! 감사합니다 :)

sjhello committed:
docs: 용어 사전 만들기

sjhello committed:
docs: 배달 대행사 영문명 수정

kyucumber commented on README.md:
```diff
@@ -94,10 +94,95 @@ docker compose -p kitchenpos up -d
+### 메뉴그룹
+| 한글명     | 영문명               | 설명                          |
+|---------|-------------------|-----------------------------|
+| 메뉴그룹    | Menu Group        | 메뉴들을 특정한 기준대로 모은 `메뉴의 집합`이다 |
```
메뉴 그룹, 상품, 메뉴 등에 대한 명확한 예시를 추가해주시면 용어 이해에 더 도움이 될 것 같네요!

kyucumber commented on README.md:
```diff
@@ -94,10 +94,95 @@ docker compose -p kitchenpos up -d
+| 한글명      | 영문명               | 설명                                                  |
+|----------|-------------------|-----------------------------------------------------|
+| 메뉴       | Menu              | 상품을 키친포스에서 팔기 위해 등록하는 단위를 말하며 메뉴는 상품으로 구성되어 있다      |
+| 메뉴 이름    | Menu name         | 메뉴의 이름을 말한다. ex> 후라이드 콜라 세트                         |
+| 메뉴 가격    | Menu Price        | 메뉴 가격                                               
+| 메뉴 구성 상품 | Menu Product      | 메뉴 구성 상품은 상품 정보와 수량으로 구성된다                          |
+| 메뉴 노출 상태 | Menu Displayed    | 메뉴의 노출 상태를 말한다. 메뉴의 노출 상태는 `노출됨`, `숨김`이 있다 |
```
메뉴가 가지는 프로퍼티와 행위를 잘 정리해주셨네요.

메뉴 뿐만 아니라 다른 도메인들 하위에 용어들이 너무 많아지면 용어 사전의 가독성이 떨어질 것 같아요. 만약 그런 부분이 우려되신다면 프로퍼티와 행위를 분리해서 정리해도 좋을 것 같다는 생각이 드네요.

kyucumber commented on README.md:
```diff
@@ -94,10 +94,95 @@ docker compose -p kitchenpos up -d
+| 메뉴 가격 변경 | Change Menu Price | 메뉴의 가격을 변경한다.                                       |
+| 메뉴 목록 조회 | Menus             | 메뉴의 목록을 확인한다.                                       |
+| 메뉴 전시    | Display Menu      | 메뉴를 노출 시킨다.                                         |
+| 메뉴 숨김    | Hide Menu         | 메뉴를 숨긴다.                                   |
```
메뉴는 어떤 경우 숨겨질까요? 메뉴 도메인을 잘 이해하기 위해 어떤 경우 숨김 메뉴가 되는지 설명을 추가하면 어떨까요?

1. 사장님이 숨기는 경우
2. 가격 정책에 의해 자동으로 숨겨지는 경우

kyucumber commented on README.md:
```diff
@@ -94,10 +94,95 @@ docker compose -p kitchenpos up -d
+| 한글명          | 영문명                   | 설명                                              |
+|--------------|-----------------------|-------------------------------------------------|
+| 매장 주문        | Eat In Order          | 키친포스의 테이블에서 식사를 하는 주문이다.                        |
+| 매장 주문의 상태    | Eat In Order Status   | 매장 주문의 상태는 `대기`, `수락`, `음식 제공됨`, `완료` 이다.       |
```
주문 타입 별 상태 흐름을 잘 표현해주셔서 한눈에 잘 이해가 되네요 👍

kyucumber commented on README.md:
```diff
@@ -94,10 +94,95 @@ docker compose -p kitchenpos up -d
+| 한글명      | 영문명               | 설명                                                  |
+|----------|-------------------|-----------------------------------------------------|
+| 메뉴       | Menu              | 상품을 키친포스에서 팔기 위해 등록하는 단위를 말하며 메뉴는 상품으로 구성되어 있다      |
+| 메뉴 이름    | Menu name         | 메뉴의 이름을 말한다. ex> 후라이드 콜라 세트                         |
```
메뉴 및 상품의 이름에는 비속어가 들어갈 수 없습니다. 해당 부분도 설명에 들어가면 어떨까요?

kyucumber commented on README.md:
```diff
@@ -94,10 +94,95 @@ docker compose -p kitchenpos up -d
+| 포장 주문 음식 제공됨 | Serve Take Out Order    | 손님에게 음식을 제공한다. 주문의 상태가 `수락`에서 `음식 제공됨` 변경된다.  |
+| 포장 주문 완료     | Complete Take Out Order | 포장 주문이 완료된다. 주문의 상태가 `음식 제공됨`에서 `완료`로 변경된다.   |
+
+### 배달 주문
```
배달 주문에 포함되는 주소에 대해서도 정리하면 어떨까요?

kyucumber commented on README.md:
```diff
@@ -94,10 +94,95 @@ docker compose -p kitchenpos up -d
+| 손님 수 변경      | Chage Number Of Guest | 테이블의 손님 수를 변경한다.                    |
+| 주문 테이블 목록 조회 | OrderTables           | 주문 테이블의 목록을 확인한다.                   |
+
+### 주문
```
주문의 경우 주문 타입별로 수량에 대한 요구사항이 있습니다.

주문 도메인의 이해를 돕기 위해 관련 설명도 추가하면 좋을 것 같아요.

kyucumber commented on README.md:
```diff
@@ -94,10 +94,95 @@ docker compose -p kitchenpos up -d
+| 주문 테이블       | Order Table           | 손님들이 앉아서 식사 할 수 있는 키친포스의 테이블을 의미한다. |
+| 테이블 사용 상태    | Occupied              | 테이블의 사용 여부를 말한다.                    |
+| 테이블 착석       | Sit                   | 테이블이 `사용중인 상태`를 말한다.                  |
+| 테이블 비어있음     | Clear                 | 테이블이 `사용중이지 않는 상태`를 말한다.              |
```
테이블 비어있음을 의미하는 clear가 동사로 비우는 행위 자체를 의미하는 것 같이 느껴지네요. 영문명을 수정하는건 어떨까요?

kyucumber replied:
테이블을 비울 때, 빈 테이블로 설정할 수 없는 경우도 있을 것 같아요. 용어의 이해를 돕기 위해 추가하면 어떨까요?

kyucumber commented on README.md:
```diff
@@ -94,10 +94,95 @@ docker compose -p kitchenpos up -d
+### 주문 테이블
+| 한글명          | 영문명                   | 설명                                  |
+|--------------|-----------------------|-------------------------------------|
+| 주문 테이블       | Order Table           | 손님들이 앉아서 식사 할 수 있는 키친포스의 테이블을 의미한다. |
```
다른 도메인과 다르게 이름이나 인원 수 등의 프로퍼티에 대한 정보가 빠진 것 같아요.

kyucumber commented on README.md:
```diff
@@ -94,10 +94,95 @@ docker compose -p kitchenpos up -d
+| 배달 완료        | Complete Delivery       | 라이더가 배달을 완료한다. 주문의 상태가 `배달 시작`에서 `배달 완료`로 변경된다.                    |
+| 배달 주문 완료     | Complete Delivery Order | 배달 주문이 완료된다. 주문의 상태가 `배달 완료`에서 `완료`로 변경된다.                      |
 
 ## 모델링
```
제가 세부적인 요구사항과 같은 부분도 적으면 어떨까 하는 부분이 다소 애매하게 느껴지실수도 있을 것 같아요.

용어 사전 정리의 경우 사실 정해진 답은 없습니다. 저는 해당 부분들이 포함되면 용어를 더 잘 이해할 수 있을 것이라 생각해서 넣으면 좋겠다고 말씀드리는거라 꼭 추가하진 않아도 괜찮습니다.

kyucumber reviewed (changes requested):
안녕하세요

몇가지 코멘트 남겨두었으니 확인 부탁드립니다.

sjhello committed:
refactor: 피드백 반영

 - 예시 추가
 - 속성(프로퍼티)과 행위 분리
 - 배달 주문, 주문 테이블 용어 사전 보충
 - 주문 수량에 대한 요구사항 추가
 - 테이블 사용 상태 정리

sjhello committed:
docs: 동료와 함께 용어사전 정의

sjhello commented:
안녕하세요 리뷰어님!
동료와 함께 용어사전을 보충해보았습니다. 바쁘신 와중에도 리뷰 해주셔서 감사합니다 :)

kyucumber reviewed (approved):
용어사전 잘 정리해주셨네요.

다음 단계 진행을 위해 머지할게요!
